lightning/src/ln/functional_tests.rs

```diff
@@ -8174,6 +8174,153 @@ fn test_override_0msat_htlc_minimum() {
+
+	// Assert that node[1] has no MessageSendEvent::SendAcceptChannel in the msg_events before rejecting the inbound request
+	assert!(nodes[1].node.get_and_clear_pending_msg_events().is_empty());
+
+	let events = nodes[1].node.get_and_clear_pending_events();
+	match events[0] {
+		Event::OpenChannelRequest { temporary_channel_id, .. } => {
+			nodes[1].node.force_close_channel(&temporary_channel_id).unwrap();
+		}
+		_ => panic!("Unexpected event"),
+	}
+
+	let close_msg_ev = nodes[1].node.get_and_clear_pending_msg_events();
+	assert_eq!(close_msg_ev.len(), 1);
+
+	match close_msg_ev[0] {
+		MessageSendEvent::HandleError { ref node_id, .. } => {
+			assert_eq!(*node_id, nodes[0].node.get_our_node_id());
+		}
+		_ => panic!("Unexpected event"),
+	}
+	check_closed_event!(nodes[1], 1, ClosureReason::HolderForceClosed);
+}
+
+#[test]
+fn test_reject_funding_before_inbound_channel_accepted() {
+	// This tests that when the util::config::UserConfig.manually_accept_inbound_channels is set to
+	// true, inbound channels must to be manually accepted through the
+	// ln::channelmanager::ChannelManager::accept_inbound_channel function by the node operator
+	// before the counterparty sends a `FundingCreated` message. If a `FundingCreated` message is
+	// received before the channel is accepted, it should be rejected and the channel should be
+	// closed.
+	let mut manually_accept_conf = UserConfig::default();
+	manually_accept_conf.manually_accept_inbound_channels = true;
+	let chanmon_cfgs = create_chanmon_cfgs(2);
+	let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+	let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, Some(manually_accept_conf.clone())]);
+	let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+	nodes[0].node.create_channel(nodes[1].node.get_our_node_id(), 100000, 10001, 42, Some(manually_accept_conf)).unwrap();
+	let res = get_event_msg!(nodes[0], MessageSendEvent::SendOpenChannel, nodes[1].node.get_our_node_id());
+	let temp_channel_id = res.temporary_channel_id;
+
+	nodes[1].node.handle_open_channel(&nodes[0].node.get_our_node_id(), InitFeatures::known(), &res);
+
+	// Assert that node[1] has no MessageSendEvent::SendAcceptChannel in the msg_events before rejecting the inbound request
+	assert!(nodes[1].node.get_and_clear_pending_msg_events().is_empty());
+
+	// Clear the Event::OpenChannelRequest event without responding to the request.
+	nodes[1].node.get_and_clear_pending_events();
+
+	// Get the `AcceptChannel` message of nodes[1] without calling the
+	// ChannelManager::accept_inbound_channel function, which generates the
+	// MessageSendEvent::SendAcceptChannel event. This is passed to nodes[0] handle_accept_channel
+	// function, which is required in order for the create_funding_transaction function not to
+	// crash when nodes[0] is passed to it.
+	{
+		let mut channel_state_lock = nodes[1].node.channel_state.lock().unwrap();
```

Comment:
Note you can use the `get_channel_ref` macro here to do most of the work.

Reply:
Great thanks! Hadn't found that one.